packages/agreement-process/src/services/agreementUpgradeProcessor.ts

```diff
@@ -97,19 +108,33 @@ export async function createUpgradeOrNewDraft({
       },
     };
 
+    const contract = await contractBuilder.createContract(
+      authData.selfcareId,
+      agreement.data,
+      eservice,
+      consumer,
+      producer
+    );
+
+    const upgradedWithContract: Agreement = {
+      ...upgraded,
+      contract,
+    };
+
```

Comment:
```suggestion
```
We can avoid declaring these two consts by creating the contract directly in `upgraded`

Reply:
The contract builder needs the updated agreement in order to create the right contract, so I need first to craft the updated agreement and then pass it to the contract builder -- so, you made me notice that this is a bug because I am passing `agreement.data` instead of `upgraded` 🥲  Fixing it in the next commit